internal/cmd/builds/list.go

```diff
@@ -0,0 +1,200 @@
+	},
+	Title: table.TitleOptionsDefault,
+}
+
+func ListCommand() *cobra.Command {
+	var out string
+	var page int
+	var size int
+	var status string
+	var buildSource string
+	var nameFilter string
+
+	cmd := &cobra.Command{
+		Use: "list",
+		Aliases: []string{
+			"ls",
+		},
+		Short:        "Returns the list of builds",
+		SilenceUsage: true,
+		PreRun: func(cmd *cobra.Command, _ []string) {
+			tracker := usage.DefaultClient
+
+			go func() {
+				tracker.Collect(
+					cmds.FullName(cmd),
+					usage.Flags(cmd.Flags()),
+				)
+				_ = tracker.Close()
+			}()
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			if page < 0 {
+				return errors.New("invalid page")
+			}
+			if size < 0 {
+				return errors.New("invalid size")
+			}
+			if out != JSONOutput && out != TextOutput {
+				return errors.New("unknown output format")
+			}
+			var isStatusValid bool
+
+			stat := build.Status(status)
+			for _, s := range build.AllStates {
+				if s == stat {
+					isStatusValid = true
+					break
+				}
+			}
+
+			if status != "" && !isStatusValid {
+				strs := make([]string, len(build.AllStates))
+				for i, n := range build.AllStates {
+					strs[i] = string(n)
+				}
+				return fmt.Errorf("unknown status. Options: %s", strings.Join(strs, ", "))
+			}
+
+			src := build.Source(buildSource)
+			if src != build.SourceRDC && src != build.SourceVDC {
+				return errors.New("invalid build resource. Options: vdc, rdc")
+			}
+
+			return list(cmd.Context(), out, page, size, stat, build.Source(buildSource), nameFilter)
+		},
+	}
+	flags := cmd.PersistentFlags()
+	flags.StringVar(&buildSource, "source", "", "Build source from saucelabs. Options: vdc, rdc. Required.")
+	flags.StringVarP(&out, "out", "o", "text", "Output format to the console. Options: text, json.")
+	flags.IntVarP(&page, "page", "p", 0, "Page for pagination. Default is 0.")
+	flags.IntVarP(&size, "size", "s", 20, "Per page for pagination. Default is 20.")
+	flags.StringVarP(&nameFilter, "name", "n", "", "Filter builds by name. Must match full build name.")
+	flags.StringVar(&status, "status", "", "Filter builds using status. Options: passed, failed, error, complete, in progress, queued.")
+
+	return cmd
+}
+
+func list(ctx context.Context, format string, page int, size int, status build.Status, source build.Source, name string) error {
+	user, err := userService.User(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get user: %w", err)
+	}
+
+	opts := build.ListBuildsOptions{
+		UserID: user.ID,
+		Page:   page,
+		Size:   size,
+		Status: status,
+		Source: source,
+		Name:   name,
+	}
+
+	builds, err := buildsService.ListBuilds(ctx, opts)
+	if err != nil {
+		return fmt.Errorf("failed to get builds: %w", err)
+	}
+
+	switch format {
+	case "json":
+		if err := renderJSON(builds); err != nil {
+			return fmt.Errorf("failed to render output: %w", err)
+		}
+	case "text":
+		renderListTable(builds)
+	}
+
+	return nil
+}
+
+func renderListTable(builds []build.Build) {
```

Comment:
This seems to be used in `get.go`. Do you think it would be a good idea to move this and `defaultTableStyle` to a separate file?

Reply:
This could also probably go into a `shared` / `common` file along with the arg processing. Long term I think it would be good to dedupe all of the `defaulTableStyle` usages across the codebase